lib/request.js

```diff
@@ -32,175 +33,192 @@ var am = require('../');
 var nextRequestId = 1;
 var nextEventId = 1;
 
-var disabledRequest = { stop: function(){} };
-
-function Request( type, name, root ) {
-    this.children = [];
-    this.name = name;
-    this.type = type;
-
-    if (root === true) {
-    	var reqDomain = domain.create();
-        reqDomain.enter();
-    }
-    
-    if (process.domain) {
-        this.parent = process.domain.currentRequest;
-    }
-    this.id = nextEventId;
-    ++nextEventId;
-    if( this.parent ) {
-        this.top = this.parent.top;
-        this.parent.children.push(this);
-    } else {
-        this.top = this;
-        if (this.type) {
-        	/* If we're the top and we're a type, we get a request Id, not an event Id */
-            this.id = nextRequestId;
-            ++nextRequestId;        	
-        }
+function Request(type, name, root) {
+  this.children = [];
+  this.name = name;
+  this.type = type;
+
+  if (root === true) {
+    var reqDomain = domain.create();
+    reqDomain.enter();
+  }
+
+  if (process.domain) {
+    this.parent = process.domain.currentRequest;
+  }
+  this.id = nextEventId;
+  ++nextEventId;
+  if (this.parent) {
+    this.top = this.parent.top;
+    this.parent.children.push(this);
+  } else {
+    this.top = this;
+    if (this.type) {
+      /* If we're the top and we're a type, we get a request Id, not an event Id */
+      this.id = nextRequestId;
+      ++nextRequestId;
     }
+  }
 }
 
 Request.prototype.traceStart = function() {
-    if( !this.tracedStart ) {
-        this.tracedStart = true;
-        // we may delay tracing parent requests until later time when they exceed threshold
-        if( this.parent ) {
-            this.parent.traceStart();
-            if( this.parent.type ) {
-                parentId = this.parent.id;
-            } else {
-                parentId = this.parent.methId;
-            }
-        }
+  if (!this.tracedStart) {
+    this.tracedStart = true;
+    // we may delay tracing parent requests until later time when they exceed threshold
+    if (this.parent) {
+      this.parent.traceStart();
+      // FIXME(sam) assigning to a global parentId that appears to never be
+      // used, but I can't be sure.
```

Comment:
Just deleting this section of code which appears to do nothing makes me nervous, its possible a global `parentId` is being used to track requests... except that would almost certainly be a bug.